horse/recommendation/service.py

```diff
@@ -0,0 +1,6 @@
+import abc
+
+
+class RecommendationService(metaclass=abc.ABCMeta):
+    def recommend(self, user):
+        raise NotImplemented
```

Comment:
That looks better. Just one more thing - please mark this method as `@abstractmethod` and change body to 
`def recommend(self, user): pass`. Raising `NotImplemented` (and btw, `NotImplemented` != `NotImplementedError`) is not necessary as subclass has to define this method.